test/index.js

```diff
@@ -51,4 +51,32 @@ describe('sequentially promises', () => {
       })
   });
 
+  it('should not keep running after one of the promises throws error', (done) => {
+
+    const resolvedPromises = { p1: false, p2: false, p3: false };
+    const expectedResolvedPromises = { p1: true, p2: false, p3: false };
+
+    function updateResolvedPromises(key) {
+      return new Promise((resolve) => {
+        resolvedPromises[key] = true;
+        resolve();
+      });
+    }
+
+    const promises = [
+      () => updateResolvedPromises('p1'),
+
+      () => Promise.reject(),
+
+      () => updateResolvedPromises('p3'),
+    ];
+
+    Promise.resolve(promiseq(promises))
+    .catch(() => {
+      expect(resolvedPromises).to.deep.equal(expectedResolvedPromises);
```

Comment:
wouldn't this still pass with the current error in the code.  That is, at the time this catch block is called, p3 has not executed yet, so the expected output should have matched as well.

Reply:
@ You are absolutely right. I applied the fix in f47c74c and I believe now it works as expected.